internal/storage/storage_test.go

```diff
@@ -301,6 +334,74 @@ func TestPOSIXFail(t *testing.T) {
 	assert.NotNil(t, err, "RemoveFile worked when it should not")
 }
 
+func TestSftpFail(t *testing.T) {
+
+	testConf.Type = sftpType
+
+	// test connection
+	tmpHost := testConf.SFTP.Host
+
+	// defer func() { testConf.SFTP.Host = tmp }()
+	testConf.SFTP.Host = "nonexistenthost"
+	_, err := NewBackend(testConf)
+	assert.NotNil(t, err, "Backend worked when it should not")
+
+	var dummyBackend *sftpBackend
+	reader, err := dummyBackend.NewFileReader("/")
+	assert.NotNil(t, err, "NewFileReader worked when it should not")
+	assert.Nil(t, reader, "Got a Reader when expected not to")
+
+	writer, err := dummyBackend.NewFileWriter("/")
+	assert.NotNil(t, err, "NewFileWriter worked when it should not")
+	assert.Nil(t, writer, "Got a Writer when expected not to")
+
+	_, err = dummyBackend.GetFileSize("/")
+	assert.NotNil(t, err, "GetFileSize worked when it should not")
+
+	err = dummyBackend.RemoveFile("/")
+	assert.NotNil(t, err, "RemoveFile worked when it should not")
+	assert.EqualError(t, err, "Invalid sftpBackend")
+	testConf.SFTP.Host = tmpHost
+
+	// wrong key password
+	tmpKeyPass := testConf.SFTP.PemKeyPass
+	testConf.SFTP.PemKeyPass = "wrongkey"
+	_, err = NewBackend(testConf)
+	assert.EqualError(t, err, "Failed to parse private key, x509: decryption password incorrect")
+
+	// missing key password
+	testConf.SFTP.PemKeyPass = ""
+	_, err = NewBackend(testConf)
+	assert.EqualError(t, err, "Failed to parse private key, ssh: this private key is passphrase protected")
+	testConf.SFTP.PemKeyPass = tmpKeyPass
+
+	// wrong key
+	tmpKeyPath := testConf.SFTP.PemKeyPath
+	testConf.SFTP.PemKeyPath = "nonexistentkey"
+	_, err = NewBackend(testConf)
+	testConf.SFTP.PemKeyPath = tmpKeyPath
+	assert.EqualError(t, err, "Failed to read from key file, open nonexistentkey: no such file or directory")
+
+	defer doCleanup()
+	dummyKeyFile, err := writeName()
+	if err != nil {
+		t.Error("could not find a writable name, bailing out from test")
+
+		return
+	}
+	testConf.SFTP.PemKeyPath = dummyKeyFile
+	_, err = NewBackend(testConf)
+	assert.EqualError(t, err, "Failed to parse private key, ssh: no key found")
+	testConf.SFTP.PemKeyPath = tmpKeyPath
+
+	// worng host key
+	tmpHostKey := testConf.SFTP.HostKey
+	testConf.SFTP.HostKey = "wronghostkey"
+	_, err = NewBackend(testConf)
+	assert.ErrorContains(t, err, "Failed to start ssh connection, ssh: handshake failed: host key verification expected")
+	testConf.SFTP.HostKey = tmpHostKey
```

Comment:
Unless the server can be mocked this test should be moved to a separate test that has a real backed available.
A test suite that is skipped unless the server is running.

Reply:
This is now tested with a mock server.